### PR TITLE
fix bug when replica is bigger than one

### PR DESCRIPTION
**Pull request by ldpliu**

### CODE REVIEW

File: cmd/controller/app/server.go [hunk 1]
```diff
@@ -182,19 +182,31 @@ func Run(o *options.ControllerRunOptions, stopCh <-chan struct{}) error {
 		klog.Errorf("unable to setup gc reconciler: %v", err)
 		return err
 	}
+	go func() {
+		for {
+			select {
```
Comment (qiujian16) [inline cmd/controller/app/server.go:187]: why  you need a for and select here? 

can we just block on <-mgr.Elected() and then run anything else?

**Review by qiujian16 — COMMENTED** [review]
/lgtm

could you also change the agent in the same way?

**Comment by ldpliu** [comment]
> /lgtm
> 
> could you also change the agent in the same way?

Ok, i will log a new task to handle this

**Pull request merged**